### PR TITLE
Test derived second stage transactions

### LINKED ISSUE #147: Invalid Cancel and Unvault Emergency signatures for derivation index = 1
I think we have the generation and check of sigs off

### PR DESCRIPTION
**Pull request by darosior**
This adds a test to fix #147 . See commit messages, what we experienced was because `bitcoind` can't understand our Unvault Scripts.

Based on #156

### CODE REVIEW

**Comment by darosior** [comment]
Note there are a few worrying logs there wrt to vanished deposits when canceling, probably due to a logic mistake in #144

File: src/daemon/control.rs [hunk 1]
```diff
@@ -263,19 +263,59 @@ fn presigned_txs_list_from_outpoints(
...
+    // For each presigned transaction, append it as well as its extracted version if it's final.
     let mut tx_list = Vec::with_capacity(db_vaults.len());
     for db_vault in db_vaults {
         let outpoint = db_vault.deposit_outpoint;
 
-        let (_, unvault) = db_unvault_transaction(db_path, db_vault.id)?;
+        let (_, unvault_psbt) = db_unvault_transaction(db_path, db_vault.id)?;
+        let mut finalized_unvault = unvault_psbt.clone();
+        let unvault = VaultPresignedTransaction {
+            transaction: if finalized_unvault.finalize(&revaultd.secp_ctx).is_ok() {
```
Comment (danielabrozzoni) [inline src/daemon/control.rs:274]: nit: I haven't tested it but I think you can just use map
```
transaction: finalized_unvault.finalize(&revaultd.secp_ctx).map(|f| f.into_psbt().extract_tx())
```

Comment (darosior) [reply src/daemon/control.rs:274]: Nope because finalize modifies the tx in place and does not return it

**Comment by darosior** [comment]
Rebased on master

**Comment by danielabrozzoni** [comment]
utACK 388b99073bfc1cc10dc00b3a66eb1f030246fb4a looks good, didn't test

**Pull request merged**